js/server/tests/aql/aql-functions-string.js

```diff
@@ -57,6 +57,140 @@ function ahuacatlStringFunctionsTestSuite () {
+      assertQueryError(errors.ERROR_QUERY_FUNCTION_ARGUMENT_NUMBER_MISMATCH.code, 'RETURN TO_HEX("test", "meow", "foo", "bar")');
+    },
+
+// //////////////////////////////////////////////////////////////////////////////
+// / @brief test encodeURIcomponent
+// //////////////////////////////////////////////////////////////////////////////
+
+    testEncodeURIComponentValues: function () {
+      [ 
+        [ null, "null" ],
+        [ -13, "-13" ],
+        [ 10, "10"],
+        [ true, "true" ],
+        [ false, "false" ],
+        [ "", "" ],
+        [ "foobar", "foobar" ],
+        [ " ", "%20" ],
+        [ "?x=шеллы", "%3Fx%3D%D1%88%D0%B5%D0%BB%D0%BB%D1%8B"],
+        [ "?x=test", "%3Fx%3Dtest"],
+        [ "The quick brown fox jumps over the lazy dog", "The quick brown fox jumps over the lazy dog"],
+        [ "https://w3schools.com/my test.asp?name=ståle&car=saab", "https%3A%2F%2Fw3schools.com%2Fmy%20test.asp%3Fname%3Dst%C3%A5le%26car%3Dsaab"],
+      ].forEach(function(test) {
+        assertEqual([ test[1] ], getQueryResults('RETURN ENCODE_URI_COMPONENT(' + JSON.stringify(test[0]) + ')'), test);
+      });
+    },
+    
+    
+    testEncodeURIComponentInvalidNumberOfParameters: function () {
+      assertQueryError(errors.ERROR_QUERY_FUNCTION_ARGUMENT_NUMBER_MISMATCH.code, 'RETURN TO_ENCODE_URI_COMPONENT()');
+      assertQueryError(errors.ERROR_QUERY_FUNCTION_ARGUMENT_NUMBER_MISMATCH.code, 'RETURN TO_ENCODE_URI_COMPONENT("test", "meow")');
+      assertQueryError(errors.ERROR_QUERY_FUNCTION_ARGUMENT_NUMBER_MISMATCH.code, 'RETURN TO_ENCODE_URI_COMPONENT("test", "meow", "foo")');
+      assertQueryError(errors.ERROR_QUERY_FUNCTION_ARGUMENT_NUMBER_MISMATCH.code, 'RETURN TO_ENCODE_URI_COMPONENT("test", "meow", "foo", "bar")');
+    },
+
+// //////////////////////////////////////////////////////////////////////////////
+// / @brief test UUID
+// //////////////////////////////////////////////////////////////////////////////
+
+    testUUIDValues: function () {
+        assertMatch(/^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/, getQueryResults('RETURN UUID()'));
+    }
+
+    testUUIDInvalidNumberOfParameters: function () {
+      assertQueryError(errors.ERROR_QUERY_FUNCTION_ARGUMENT_NUMBER_MISMATCH.code, 'RETURN UUID("test")');
+      assertQueryError(errors.ERROR_QUERY_FUNCTION_ARGUMENT_NUMBER_MISMATCH.code, 'RETURN UUID("test", "meow")');
+      assertQueryError(errors.ERROR_QUERY_FUNCTION_ARGUMENT_NUMBER_MISMATCH.code, 'RETURN UUID("test", "meow", "foo")');
+      assertQueryError(errors.ERROR_QUERY_FUNCTION_ARGUMENT_NUMBER_MISMATCH.code, 'RETURN UUID("test", "meow", "foo", "bar")');
+    },
+
+// //////////////////////////////////////////////////////////////////////////////
+// / @brief test Soundex
+// //////////////////////////////////////////////////////////////////////////////
+    testToSoundexValues: function () {
+      [ 
+        [ null, "N400" ],
```

Comment:
Is this really desired? Shouldn't non-string types produce an empty string or `null` as result?

Reply:
Good point...
The soundex function should first stringify its input argument. 
`null` converted to string in AQL using `TO_STRING()` is the empty string (""). 
I don't know what the soundex value for an empty string is, so we need to find out.
I think `N400` is the soundex value for the input string `"null"`.
Probably we need to make exceptions for empty/non-expected input values or for non-string input values, and either return `null` or a special value such as `0000`? How have other databases (e.g. MySQL) solved this?